vpn/client.go

```diff
@@ -195,10 +226,14 @@ func (c *Client) LocalAddr() net.Addr {
 	return addr
 }
 
-// TODO(ainghazal): should get the remote _tunnel_ ip addr somehow
+// RemoteAddr returns the address of the tun interface of the tunnel gateway.
 func (c *Client) RemoteAddr() net.Addr {
-	logger.Errorf("client.RemoteAddr() not implemented")
-	return nil
+	if c.tunInfo == nil {
+		addr, _ := net.ResolveIPAddr("ip", "")
+		return addr
+	}
+	addr, _ := net.ResolveIPAddr("ip", c.tunInfo.gw)
```

Comment:
Is the `ResolverIPAddr` function going to use the DNS if the `c.tunInfo.gw` contains a domain? In such a case, is there a more direct function we can use to init a `&net.IPAddr{}`?

Reply:
Maybe something like:

```
addr := &net.IPAddr{}
if c.tunInfo != nil {
  if ip := net.ParseIP(c.tunInfo.gw); ip != nil {
    addr.IP = ip
  }
  return addr
}
```

This code would break in case we have an IPv6 interface scope, though.

In such a case we can use `net/netip.ParseAddr` so it becomes

```
if c.tunInfo == nil {
  return &net.IPAddr{} // XXX: I'd rather document this or define a (net.Addr, bool) or (net.Addr, error) API
}
addr, _ := netip.ParseAddr(c.tunInfo.gw)
return addr // XXX: note that addr will most likely be nil here
```

Reply:
So, maybe it would be better to just return `nil` on failure and also return a `bool` indicating if the result is OK?

Reply:
the net.Conn interface expects a net.Addr, _if known_, so changing as per suggestions (I agree it was not the best option to parse the addr).

Reply:
trying this a bit more, I see `netip.Addr` doesn't implement 'net.Addr` (it's missing `Network` method). 

first suggestion seems good to me though, but I don't get why would it break w/ IPv6? https://pkg.go.dev/net#ParseIP

Reply:
it does not know interface scope IDs such as `fe80::1:2:3:4%eth0` but I think they are only used for link local IP addresses, so probably we should be fine for quite a long time here 😬